src/tools/rbd/action/MirrorPool.cc

```diff
@@ -1666,6 +1723,105 @@ int execute_status(const po::variables_map &vm,
     utils::get_mirror_peer_mirror_uuids_to_names(mirror_peers,
                                                  &peer_mirror_uuids_to_name);
 
+    std::map<std::string, librbd::mirror_group_global_status_t> mirror_groups;
+    r = rbd.mirror_group_global_status_list(io_ctx, "", 1024, &mirror_groups);
+    if (r < 0) {
+      std::cerr << "Failed to list mirror groups: " << r << std::endl;
+    }
+    librbd::mirror_group_site_status_t local_status_group;
+    if (total_groups > 0) {
+      if (formatter != nullptr) {
+        formatter->open_array_section("groups");
+        for (auto &[group_id, group_status] : mirror_groups) {
+          int local_site_group_r = utils::get_local_mirror_group_status(
+          group_status, &local_status_group);
+          group_status.site_statuses.erase(
+            std::remove_if(group_status.site_statuses.begin(),
+                          group_status.site_statuses.end(),
+                          [](auto& group_status) {
+                return (group_status.mirror_uuid ==
+                          RBD_MIRROR_GROUP_STATUS_LOCAL_MIRROR_UUID);
+              }),
+            group_status.site_statuses.end());
+          formatter->open_object_section("group");
+          formatter->dump_string("name", group_status.name);
+          formatter->dump_string("global_id",group_status.info.global_id);
+          if (local_site_group_r >= 0) {
+            formatter->dump_string("state", utils::mirror_group_site_status_state(
+              local_status_group));
+            formatter->dump_string("description", local_status_group.description);
+            formatter->dump_string("last_update", utils::timestr(
+              local_status_group.last_update));
+          }
+          formatter->open_array_section("peer_sites");
+          for (auto& status : group_status.site_statuses) {
+            formatter->open_object_section("peer_site");
+
+            auto name_it = peer_mirror_uuids_to_name.find(status.mirror_uuid);
+            formatter->dump_string("site_name",
+              (name_it != peer_mirror_uuids_to_name.end() ? name_it->second : ""));
+            formatter->dump_string("mirror_uuids", status.mirror_uuid);
```

Comment:
Should this be "mirror_uuid" instead of "mirror_uuids" , we're displaying a single UUID ?